robot/rover/internal_comms/include/CommandCenter.h

```diff
@@ -2,39 +2,79 @@
 #define INTERNAL_COMMS_COMMANDCENTER_H
 //
 // Created by cedric on 2020-10-14.
+// And mostly written by Tim
 //
 
 #include "Arduino.h"
+#include <etl/queue.h>
 
 namespace internal_comms
 {
+   typedef struct {
+       uint8_t commandID; // ID of command
+       uint8_t* rawArgs; // Byte array with the bytes of the arguments
+       uint16_t rawArgsLength; // Number of bytes in the rawArgs array
+       bool isValid; // Whether the command is valid
+   } Command;
+
     typedef struct {
-        uint8_t commandID; // ID of command
-        uint8_t* rawArgs; // Byte array with the bytes of the arguments
-        uint16_t rawArgsLength; // Number of bytes in the rawArgs array
-        bool isValid; // Whether the command is valid
-    } Command;
+        uint8_t messageID;
+        uint16_t rawArgsLength;
+        uint8_t* rawArgs;
+    } Message;
 
     class CommandCenter {
         public:
 
             /**
-             * Executes the command with the commandID with the raw arguments
-             * @param commandID The ID of the command to execute
-             * @param rawArgs The raw bytes consisting of the arguments
-             */
-            virtual void executeCommand(const uint8_t commandID, const uint8_t* rawArgs, const uint8_t rawArgsLength) = 0;
+            * Executes the command with the commandID with the raw arguments
+            * @param commandID The ID of the command to execute
+            * @param rawArgs The raw bytes consisting of the arguments
+            */
+            virtual void executeCommand(const uint8_t commandID,
+                                        const uint8_t* rawArgs, 
+                                        const uint8_t rawArgsLength) = 0;
+
+            /**
+            * Reads serial to next command
+            * and returns a command struct
+            *
+            * @return returns a command struct
+            */
+            static Command* processCommand() ;
 
             /**
-             * Reads serial the next command
-             * and returns a command struct
-             *
-             * @return returns a command struct
+            * Queues messages so that they are ready to be sent
+            * allows teensy.
+            * Currently we do not do anything if the queue fills up.
+            * Hopefully we can find a decent queue size that will never fill.
+            * @message reference to message struct
+            */
+            void queueMessage(Message& message);
+
+            /**
+             * @param cmdID command ID of message to send
+             * @param argSize number of arguments in the message
+             * @param args contains raw bytes of arguments
+             * @return returns a message struct
              */
-            Command* processCommand() const;
+            Message* createMessage(int cmdID, int argSize, byte *args);
```

Comment:
it should be consistent with `executeCommand` the names of the parameters are commandID and rawArgsLength. I dont care which one you change but make sure its consistent.